### PR TITLE
nRF Desktop: simplify Bluetooth configuration part 2

### PR DESCRIPTION
**Pull request by kapi-no**

### CODE REVIEW

File: applications/nrf_desktop/Kconfig.ble [hunk 1]
```diff
@@ -55,4 +62,14 @@ config BT_CONN_PARAM_UPDATE_TIMEOUT
 
 endif
 
+if DESKTOP_BT_CENTRAL
+
+config BT_GATT_CLIENT
+	default y
+
+config BT_GATT_DM
+	default y
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/Kconfig.ble:71]: Maybe modules that use these options could select them?

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: applications/nrf_desktop/Kconfig.ble [hunk 2]
```diff
@@ -20,6 +20,13 @@ config DESKTOP_BT_PERIPHERAL
 	select DESKTOP_BT
 	select BT_PERIPHERAL
 
+config DESKTOP_BT_CENTRAL
```
[the remark this answers is not in the document]
Comment (kapi-no) [reply applications/nrf_desktop/Kconfig.ble:23]: Are we sure that we want to have such a constraint? It is not uncommon to be both Peripheral and Central  (e.g. DESKTOP_ROLE_HID_RELAY). I think that some Audio roles require both Central and Peripheral capabilities (e.g. Scan Delegator, Broadcast Assistant)

File: applications/nrf_desktop/Kconfig.ble (hunk 1 above)
[the remark this answers is not in the document]
Comment (kapi-no) [reply applications/nrf_desktop/Kconfig.ble:71]: I thought that it should be the other way around:

1. nRF Desktop Role determines the high-level use case of the application configuration
2. Based on the high-level role, we may want to add Bluetooth connectivity for Bluetooth devices (DESKTOP_BT_CENTRAL / DESKTOP_BT_PERIPHERAL)
3. DESKTOP_BT_* selects modules and configuration to fulfill its role in the Bluetooth scope

[the remark this answers is not in the document]
Comment (kapi-no) [reply applications/nrf_desktop/Kconfig.ble:71]: I could introduce here independent roles DESKTOP_BT_GATT_CLIENT and DESKTOP_BT_GATT_SERVER that would be selected by some higher-level role. For example:

```
config DESKTOP_ROLE_HID_BT_PERIPHERAL
  select DESKTOP_BT_PERIPHERAL
  select DESKTOP_BT_GATT_SERVER
```

[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/Kconfig.ble:71]: Then maybe we should have a single Kconfig option for enabling Bluetooth connectivity for nRF Desktop device (this config could enable the needed Bluetooth modules depending on the fact if device is HID peripheral or HID dongle)? In general HID peripheral works only as a Bluetooth peripheral and HID dongle works only as a Bluetooth central.

[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/Kconfig.ble:71]: In the future, we could introduce new nRF Desktop roles and expand enabled Bluetooth configuration to match new use-cases.

[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/Kconfig.ble:71]: Regarding selecting Kconfigs, I meant that we could select given functionality in Kconfig that enables module that uses the functionality (e.g. select `CONFIG_BT_GATT_DM` in BLE discovery module Kconfig). Some of the options (like BT_PERIPHERAL or BT_CENTRAL) do not belong directly to any application module, so we should still configure them in global Kconfig (as it's done now).

File: applications/nrf_desktop/Kconfig.ble (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/Kconfig.ble:23]: Then we could consider using only one option with prompt that enables Bluetooth connectivity (see my comment above). The exact set of enabled modules would still depend on nRF Desktop role, so we could still introduce new modules/options related to role selected by the user.

File: applications/nrf_desktop/src/modules/Kconfig.ble_passkey [hunk 3]
```diff
@@ -13,6 +13,9 @@ menuconfig DESKTOP_BLE_ENABLE_PASSKEY
 
 if DESKTOP_BLE_ENABLE_PASSKEY
 
+config BT_SMP_ENFORCE_MITM
+	default n
```
[the remark this answers is not in the document]
Comment (kapi-no) [reply applications/nrf_desktop/src/modules/Kconfig.ble_passkey:17]: Added

File: applications/nrf_desktop/Kconfig.ble (hunk 1 above)
[the remark this answers is not in the document]
Comment (kapi-no) [reply applications/nrf_desktop/Kconfig.ble]: Changed it according to your comments and the discussion from the meeting

File: applications/nrf_desktop/Kconfig.ble (hunk 2 above)
[the remark this answers is not in the document]
Comment (kapi-no) [reply applications/nrf_desktop/Kconfig.ble]: Addressed

**Review by MarekPieta — APPROVED** [review]

**Review by pdunaj — APPROVED** [review]

**Pull request merged**